### PR TITLE
Add ArcGis geocoding service

### PR DESCRIPTION
**Pull request by kuzvac**
Added ArcGis geocoding service which support geocode, reverse geocode and autocomplete.
Docs can be found on https://developers.arcgis.com/rest/geocode/api-reference/geocoding-find-address-candidates.htm and https://developers.arcgis.com/rest/geocode/api-reference/geocoding-reverse-geocode.htm
Have a good day! :)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ var L = require('leaflet'),
 	What3Words = require('./geocoders/what3words'),
 	Google = require('./geocoders/google'),
 	Photon = require('./geocoders/photon'),
-	Mapzen = require('./geocoders/mapzen');
+	Mapzen = require('./geocoders/mapzen'),
+	AcrGis = require('./geocoders/acrgis');
```
Comment (perliedman) [inline src/index.js]: Looks like a typo here, you're referring the variable `ArcGis` below. Did you build and test the code to check that it works?

**Comment by perliedman** [comment]
Please rename the file `acrgis.js` to `arcgis.js`, and I think there's a related problem that I mentioned as a code comment above.

**Comment by kuzvac** [comment]
Ah, yes, thank you. Now it's working properly.

**Comment by perliedman** [comment]
Can you squash these commits into one, to make the history look a bit cleaner?

**Comment by kuzvac** [comment]
I do not know how to do it correctly :(

**Comment by perliedman** [comment]
No worries, it's (like many things with git) a bit tricky the first time.

I'm sure there's like a million tutorials on how to do it, but here's a short summary:

The idea of "squashing" commits is to bunch together a number of commits into a single one. Often to avoid a lot of "fixed this/that" commits that are just confusing in the repo history.

To do it, you use the command `git rebase -i`, which means "interactive rebase". You need to give the commit hash of the commit you want to "start" your rebase at. In your case, you probably want:

```
git rebase -i b83b2cfbee69e4bf48f723f5b24e5143ba89760b
```

This will bring up an editor, where all commits following b83b2cfbee69e4bf48f723f5b24e5143ba89760b are listed. By default, they will all have the command `pick` in front of them. Commits with `pick` in front of them will not be altered, but if you change `pick` into `squash` (or `fixup`), will squash this commit together the previous commit.

After you save and exit the editor, git will perform the modifications. Beware that it's actually possible to _remove_ commits from the history in this interface, if you remove lines from the file.

After squashing, you can push your changes again, but you will have to use `git push -f`, to force changes to previous commits.

Feel free to try it out, it's a good thing to know about. If you don't want to, no worries, just tell me and I'll try to find the time and fix up the last bits.

**Comment by kuzvac** [comment]
Thank you for help, i try to squash commit a bit later.

**Comment by perliedman** [comment]
GitHub now allows squashing as part of accepting a PR, so this was solved automatically.

Again, thanks for contributing this code! :smile_cat:

**Comment by kuzvac** [comment]
Community FTW! :)

**Pull request merged**